features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -466,7 +468,11 @@ private Date getNextRunDate(String cronSchedule, Date lastRun) {
     }
 
     private static DeviceConfigDTO createDeviceConfigDto(DeviceConfig deviceConfig) {
+        // TODO: Handle different character encodings and also binary data
         try {
+            String config =
+                deviceConfig.getConfigType() == null ? "" : new String(deviceConfig.getConfig(), StandardCharsets.UTF_8);
```

Comment:
why check for `deviceConfig.getConfigType()` ? 
Also it is `required` field so it will always be non-null

Reply:
Can be updated to 
```suggestion
               String encoding = deviceConfig.getEncoding != null ? deviceConfig.getEncoding : Charset.defaultCharset().name();
                String config = new String(deviceConfig.getConfig(), Charset.forName(encoding));
```

Reply:
`getConfigType` was a typo on my part! Should have been `getConfig`. Also, I agree with your suggestion, will change

Reply:
Updated